bdb/phys_rep_lsn.c

```diff
@@ -295,7 +295,13 @@ LOG_INFO find_match_lsn(void *in_bdb_state, cdb2_hndl_tp *repl_db,
                     info.file = match_file;
                     info.offset = match_offset;
                     info.size = blob_len;
-                    info.gen = *(int64_t *)cdb2_column_value(repl_db, 2);
+                    int64_t *gen = (int64_t *)cdb2_column_value(repl_db, 2);
+                    if (gen == NULL) {
+                        info.gen = 0;
+                        logmsg(LOGMSG_ERROR, "physreps require elect-highest-committed-gen source\n");
```

Comment:
Should we exit at this point?

Reply:
I was imagining situations where the source database was accidentally rolled back to a version which doesn't include generation numbers in the commit records.  If we don't exit, clients against these physreps would still maintain read access at that point in time.  I'm not sure which is better, but I'll add a tunable which would force an exit here.